Every 2 hours "invalid status code from server"

Every 2 hours I see this error in the log:
2025/12/24 04:11:02 [INFO] Re-authenticating to UniFi Controller: https://192.168.0.1/
2025/12/24 04:11:03 [ERROR] metric fetch failed: unifi.GetSites(): controller: https://192.168.0.1/proxy/network/api/stat/sites: 401 Unauthorized: invalid status code from server: error initializing inputs

I guess, that this may be the fault of the server, but Unpoller should do a retry. Otherwise, I am missing the data from this minute.

UDM-Pro 4.4.6. Network 9.5.21. Unpoller with Prometheus.

Thanks!

--- a/pkg/inputunifi/collector.go
+++ b/pkg/inputunifi/collector.go
@@ -80,9 +80,16 @@ func (u *InputUnifi) collectController(c *Controller) (*poller.Metrics, error) {
 	if err != nil {
 		u.Logf("Re-authenticating to UniFi Controller: %s", c.URL)
 
-		if err := u.getUnifi(c); err != nil {
-			return metrics, fmt.Errorf("re-authenticating to %s: %w", c.URL, err)
+		if authErr := u.getUnifi(c); authErr != nil {
+			return metrics, fmt.Errorf("re-authenticating to %s: %w", c.URL, authErr)
 		}
+
+		// Brief delay to allow controller to process new authentication
+		time.Sleep(500 * time.Millisecond)
+
+		// Retry the poll after successful re-authentication
+		u.LogDebugf("Retrying poll after re-authentication: %s", c.URL)
+		metrics, err = u.pollController(c)
 	}
 
 	return metrics, err

